spec/avro/builder_spec.rb

```diff
@@ -1029,6 +1027,166 @@
     end
   end
 
+  context "array of unions" do
+    subject do
+      described_class.build do
+        record :unions do
+          required :a, :array, items: union(:string, :int)
+        end
+      end
+    end
+    let(:expected) do
+      {
+        type: :record,
+        name: :unions,
+        fields: [
+          {
+            name: :a,
+            type: {
+              type: :array, items: [:string, :int]
+            }
+          }
+        ]
+      }
+    end
+    it { is_expected.to be_json_eql(expected.to_json) }
+  end
+
+  context "array of unions via method" do
+    subject(:schema_json) do
+      described_class.build do
+        record :unions do
+          required :a, array(union(:string, :int))
+        end
+      end
+    end
+    let(:expected) do
```

Comment:
No, I decided not to support this syntax, so it can be removed. Same for the other example you flagged.